server/src/main/java/org/elasticsearch/transport/RemoteClusterConnection.java

```diff
@@ -217,7 +222,7 @@ public RemoteConnectionInfo getConnectionInfo() {
             connectionStrategy.getModeInfo(),
             initialConnectionTimeout,
             skipUnavailable,
-            REMOTE_CLUSTER_PROFILE.equals(remoteConnectionManager.getConnectionProfile().getTransportProfile())
+            hasCredentials
```

Comment:
No need to store `hasCredentials` since we can check it again with `credentialsManager`. 
```suggestion
            remoteConnectionManager.getCredentialsManager().hasCredentials(clusterAlias)
```